src/components/album/AlbumsList.js

```diff
@@ -22,41 +24,51 @@ import {
 
 const AlbumsList = () => {
   const [albums, setAlbums] = useState([]); // for albums
-  const [limit, setLimit] = useState(1); // for pagination
+  const limit = 12;
+  // const [limit, setLimit] = useState(12); // for pagination
   const [searchType, setSearchType] = useState('albumName'); //default value for select input
   const [searchTerm, setSearchTerm] = useState(''); // for search input filed value
   const [message, setMessage] = useState(''); // for not found message
   const [errorMessage, setErrorMessage] = useState(''); // for error message
   const [wishListId, setWishListId] = useState();
+  const [currentPage, setCurrentPage] = useState(1); // Track current page
+  // const [lastSearchTerm, setLastSearchTerm] = useState('');
 
   //make an API call with search values to backend and return the result
-  const fetchAlbums = async (searchType, searchTerm, limit) => {
-    try {
-      const response = await axios.get(
-        `${process.env.REACT_APP_API_BASE_PATH}/albums/filter`,
-        {
-          params: {
-            limit,
-            [searchType]: searchTerm,
-          },
+  const fetchAlbums = useCallback(
+    async (searchType, searchTerm) => {
+      try {
+        const offset = (searchTerm === '' ? currentPage - 1 : 0) * limit;
```

Comment:
Ohh nice, I like the idea of this approach!

We do again have an issue where, if I have a search term, then pagination doesn't work:

Eg:


https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/07def0b6-b71e-4dd3-89ef-da82a4112e21


*   Click next, `currentPage` is updated to `2`. `searchTerm` is still `"L"`. `offset = ("L" === "" ? 2 -1 : 0)*12 = (false ? 1 :0)*12 = (0) * 12 = 0`

Reply:
I think we can instead always, update the offset based on the current page, regardless of the searchTerm value:

```suggestion
        const offset = (currentPage - 1) * limit;
```

Reply:
I was not able to play the video.
Anyway I did this fix and unfortunately after that the search is stoped working at all from all pages except first. So, if you are on the first page and you are searching for any albums or artists(from any page) - it works, but the search from other pages - not working at all in the case. 
So, should I apply this fix, or it will be better to leave it as is?

Reply:
After adding all the changes listed below, this fix began to work.